pkg/rtc/mediatrack.go

```diff
@@ -49,7 +49,7 @@ type MediaTrack struct {
 	audioLevelMu sync.RWMutex
 	audioLevel   *AudioLevel
 
-	receiver        sfu.Receiver
+	receiver        sfu.TrackReceiver
```

Comment:
why not continue to use sfu.Receiver, need to cast to sfu.Receiver in some place

Reply:
Thank you @cnderrauber . I want to reuse the receiver parts from `MediaTrack` also with the remote media track. So, using the interface and casting as necessary for now. That will allow to sharpen the interfaces further and make re-use better.

This PR will have conflicts with your session migration stuff @cnderrauber . So, was waiting for you to merge and then resolve conflicts and ask for review. But, you already did it :-) But, I will wait for you to merge, resolve conflicts and take it from there.

This allows full re-use of uptrackmanager so that remote participant becomes a lot smaller.

More to come...

@davidzhao Might as well ask you for review also now as Jie has already looked at it :-). But, will resolve conflicts after session migration and then only merge this.

Reply:
Sounds good, Thanks!